File: week1/calculator.java
import java.util.Scanner;

public class calculator {
    public static void main(String[] args) {
    Scanner sc = new Scanner(System.in);
    char op;
    double n1, n2, result;
    
    opyt = sc.nextLine();  // tackle from buffer error 
    System.out.println("Enter an operator (+, -, *, /): ");
    op = sc.nextLine();  
    
    // Prompt the user to enter two numbers
    System.out.println("Enter the first number: ");
    n1=sc.nextInt();
    
    System.out.println("Enter the second number: ");
    n2 = sc.nextInt();
    
    // Perform the operation based on the operator
    switch (op) {
        case '+':
            result = n1 + n2;
            printf("Result--> %.2lf\n", result);
            break;
        case '-':
            result = n1 - n2;
            printf("Result--> %.2lf\n", result);
            break;
        case '*':
            result = n1 * n2;
            printf("Result:-->%.2lf\n", result);
            break;
        case '/':
            
            if (n2 != 0) {
                result = n1 / n2;
                System.out.println("Result--> %.2lf\n", result);
            } else {
                System.out.println("Error !!! Division by zero is not Define\n");
            }
            break;
        default:
        System.out.println("Error!!! Invalid operator.\n");
            break;
    }

    

    }
}

File: calculator.c
#include <stdio.h>
int main()
{
    float a, b;
    float result;
    char op, qw;
    printf("Enter the two number: \n");
    scanf("%f%f", &a, &b);
    printf("Enter the operation ");
    scanf("%c", &qw);
    scanf("%c", &op);
    switch (op)
    {
    case '+':
        result = a + b;
        printf("THe sum of %2.f and %2.f --> %2.f ", a, b, result);
        break;
    case '-':
        result = (a - b);
        printf("The substraction of %2.f and %2.f -->%2.f", a, b, result);
        break;
    case '*':
        result = (a * b);
        printf("The multiplication of %2.f and %2.f -->%2.f", a, b, result);
        break;
    case '/':
    if (b != 0) {
                result = a / b;
                printf("Result: %.2lf\n", result);
            } else {
                printf("Error: Division by zero is not allowed.\n");
            }
            break;

    default: printf("The invalid operation!! \n");
    break;
    }
    return 0;
}